packages/lit-dev-content/site/articles/article/lit-with-tailwind.md

```diff
@@ -0,0 +1,227 @@
+export default {
+  content: [
+    './src/**/*.ts'
+  ],
+  theme: {
+    extend: {},
+  },
+  plugins: [],
+}
+```
+
+Tailwind [configuration](https://tailwindcss.com/docs/configuration) can be
+modified to fit your use case. The important part is for `content` to specify
+where our component sources are so tailwind can detect which classes we have
+used.
+
+In our `postcss.config.js`, we simply want to tell postcss to use tailwind
+as a plugin:
+
+```ts
+export default {
+  plugins: {
+    tailwindcss: {}
+  }
+};
+```
+
+## Build scripts
+
+We're going to do a few steps in our build:
+
+1. Bundle our sources (TypeScript in this case, including our CSS imports)
+2. Apply tailwind styles to the build output in-place
+
+To do this, we can use an npm script:
+
+```json
+{
+  "scripts": {
+    "build:js": "esbuild --format=esm --bundle --loader:.css=copy --outfile=bundle.js src/main.ts",
+    "build:css": "postcss -r \"./*.css\"",
+    "build": "npm run build:js && npm run build:css"
+  }
+}
+```
+
+You can see the magic here in our `build:js` script is esbuild's `--loader`
+option, which we have set to `.css=copy`. This basically means any CSS files we
+import via an ESM import statement will be copied across to the same directory
+as the esbuild JS output (`bundle.js` in our case).
+
+So if we import `my-element.css`, we will expect two files in our directory:
+
+- `my-element.css` (probably under another name, using esbuild's chunk naming)
+- `bundle.js`
+
+Finally, the `build:css` script then _replaces_ (via the `-r` flag) those CSS
+files with copies which now have the tailwind styles injected.
+
+## Run it!
+
+We now have a `build` script! You can run this via `npm run build` and should
+see two new files appear as mentioned before.
+
+## Expressions in styles (dynamic styles)
+
+You may have noticed one thing we have lost by having our CSS in external
+files: we can no longer template variables into our CSS like so:
+
+```ts
+static styles = css`
+  .foo {
+    color: ${dynamicColourDefinedInJs};
+  }
+`;
+```
+
+There are many challenges in supporting this usage and still passing it through
+tailwind. For this reason, **it is not recommended**.
+
+Instead, in places we really need to do this, we should use CSS variables or
+do the computation statically at build-time (replace values in our CSS at
+build time).
+
+To solve this with CSS variables, we can do the following:
+
+`my-element.css`:
+
+```css
+.foo {
+  color: var(--foo-color);
+}
+```
+
+`my-element.ts`:
+
+```ts
+static styles = [
+  css`
+    :host {
+      --foo-color: ${dynamicColour};
+    }
+  `,
+  myElementStyles
+];
+```
+
+The initial `styles` entry in this case is only being used for setting the
+values of some CSS variables, and never used for actual styling. We can then
+reference those variables in our external CSS files.
+
+## Example
+
+An example repository very similar to this guide is available here:
+
+https://github.com/43081j/tailwind-lit-example

```

Comment:
can we do a link that's like:

```
[Source](https://github.com/43081j/tailwind-lit-example)
[Live demo](https://githubblitz.com/43081j/tailwind-lit-example)
```

All you would also need to do is have a `start` npm command that starts a server so that we can serve it like with WDS